### PR TITLE
update getTransactionReceipt return type

### LINKED ISSUE #4819: effective gas price should be a big number.
effective gas price and other price values should be a big number.

Starting web3.js 1.7.1 you will be unable to get a receipt for [the transaction](https://etherscan.io/tx/0xeda3bf566ae21fbc7bbfcb805d73181b0037036d700d57695e28f1e9bfe0d69c)

Reason - very big gas price, number overflow

_Originally posted by @mark-b-ab in https://github.com/ChainSafe/web3.js/issues/4694#issuecomment-1058089193_

### PR DESCRIPTION
**Pull request by Hyun2**
## Description
Changed the type of `effectiveGasPrice` number to string in the object returned by the `getTransactionReceipt` function due to a number overflow issue.

Fixes #4819 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Comment by nikoulai** [comment]
Could you please update the doc [here](https://github.com/ChainSafe/web3.js/blob/832271a38e555fa62c843a05aba4eb5514f53d8a/docs/web3-eth.rst#L1427)?

**Comment by Hyun2** [comment]
> Could you please update the doc [here](https://github.com/ChainSafe/web3.js/blob/832271a38e555fa62c843a05aba4eb5514f53d8a/docs/web3-eth.rst#L1427)?

@nikoulai  thanks, updated.

**Review by nikoulai — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Review by nazarhussain — APPROVED** [review]

**Review by MitrichDot — APPROVED** [review]

File: packages/web3-core/types/index.d.ts [hunk 1]
```diff
@@ -213,7 +213,7 @@ export interface TransactionReceipt {
     contractAddress?: string;
     cumulativeGasUsed: number;
     gasUsed: number;
-    effectiveGasPrice: number;
+    effectiveGasPrice: string;
```
Comment (jdevcs) [inline packages/web3-core/types/index.d.ts:216]: This will be breaking change in 1.x

**Review by MitrichDot — APPROVED** [review]

**Comment by jdevcs** [comment]
This issue is fixed in https://github.com/ChainSafe/web3.js/pull/5157 without introducing breaking change, so closing this PR. https://github.com/ChainSafe/web3.js/pull/5157#pullrequestreview-1023233829

**Pull request closed without merging**